diag_manager/fms_diag_buffer.F90

```diff
@@ -1363,7 +1363,7 @@ subroutine add_to_buffer_2d(this, input_data)
   class(buffer2d), intent(inout) :: this !< allocated 2d buffer object
   class(*), intent(in)             :: input_data(:,:) !< 2d data array to copy into buffer
   integer            :: n1, n2 !< number of elements per dimension
-  logical            :: type_error
```

Comment:
This needs a comment and should be set on the next line if you want to initialize it to false every time
```fortran
logical            :: type_error !< ADD A COMMENT
type_error = .false.
```